### PR TITLE
Fix build-kfctl to copy the correct binary on darwin arch

### PR DESCRIPTION
**Pull request by adrian555**
Current `build-kfctl` in Makefile copies `linux` version of kfctl even built on MacOS. Fixes with querying the GOOS env from go command.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/188)

### CODE REVIEW

File: third_party/license_info.csv [hunk 1]
```diff
@@ -317,11 +317,11 @@ kubernetes-sigs/yaml,https://github.com/kubernetes-sigs/yaml/blob/master/LICENSE
...
-modernc.org/golex,https://gitlab.com/cznic/golex/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/golex/blob/master/LICENSE
-modernc.org/mathutil,https://gitlab.com/cznic/mathutil/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/mathutl/blob/master/LICENSE
-modernc.org/strutil,https://gitlab.com/cznic/strutil/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/strutil/blob/master/LICENSE
-modernc.org/xc,https://gitlab.com/cznic/xc/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/xc/blob/master/LICENSE
-heketi/utils,https://github.com/heketi/heketi/blob/master/LICENSE,Apache License 2.0,https://github.com/heketi/heketi/blob/master/LICENSE
-kr/logfmt,https://github.com/kr/logfmt/blob/master/Readme,MIT License,https://github.com/kr/logfmt/blob/master/Readme
+bitbucket.org/bertimus9/systemstat,https://bitbucket.org/bertimus9/systemstat/raw/master/LICENSE,MIT License,https://bitbucket.org/bertimus9/systemstat/raw/master/LICENSE
```
Comment (animeshsingh) [inline third_party/license_info.csv]: Is this duplicate again here?

Comment (adrian555) [reply third_party/license_info.csv]: Looks like they were carried from previous branch.

Comment (jlewi) [reply third_party/license_info.csv]: How come this PR is modifying the licenses?

Comment (adrian555) [reply third_party/license_info.csv]: I have removed the unrelated changes. Thanks @jlewi .

**Pull request merged**